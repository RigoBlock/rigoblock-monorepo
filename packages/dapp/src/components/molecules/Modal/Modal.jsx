import './Modal.scss'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

class Modal extends PureComponent {
  onClickOutside = e => {
    const { onClickOutside } = this.props
    if (!this.modalContent.contains(e.target)) {
      onClickOutside()
    }
  }

  render() {
    const { Component } = this.props
    return Component ? (
      <div className="modal-wrapper" onClick={this.onClickOutside}>
        <div
          className="modal-content"
          ref={modalContent => (this.modalContent = modalContent)}
        >
          <Component />
        </div>
      </div>
    ) : null
  }
}

Modal.propTypes = {
  Component: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
  onClickOutside: PropTypes.func
}

Modal.defaultProps = {
  Component: null,
  onClickOutside: () => {}
}

export default Modal
