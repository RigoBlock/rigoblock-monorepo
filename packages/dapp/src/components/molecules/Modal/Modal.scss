@import "../../_settings/variables.scss";

.modal-wrapper {
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  display: flex;
  overflow: hidden;
  background: $white-semi-transparent;
  align-items: center;
  justify-content: center;
  z-index: 9999;
  .modal-content {
    background: $white;
    border-radius: $dist-2xs;
    min-height: 200px;
    min-width: 200px;
    box-shadow: $box-shadow2;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
