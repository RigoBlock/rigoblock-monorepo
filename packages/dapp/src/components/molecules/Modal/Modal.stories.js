import '../../_settings/_base.scss'
import { boolean, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import Modal from './Modal'
import React from 'react'

const dummyContent = () => <div>Opened!</div>

storiesOf('Molecules/Modal', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div
      style={{
        width: '1024px',
        height: '600px',
        background:
          'linear-gradient(90deg, rgba(73,63,242,1) 0%, rgba(9,9,121,1) 35%, rgba(0,212,255,1) 100%)'
      }}
    >
      {story()}
    </div>
  ))
  .add('default', () => (
    <Modal Component={boolean('OPENED', true) ? dummyContent() : null} />
  ))
