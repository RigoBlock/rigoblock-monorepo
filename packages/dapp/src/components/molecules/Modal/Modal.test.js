import { mount, shallow } from 'enzyme'
import Modal from './Modal.jsx'
import React from 'react'
import toJson from 'enzyme-to-json'

const dummyComp = <div> hello</div>
const props = { Component: dummyComp, onClickOutside: jest.fn() }

describe('Modal component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Modal, props)))
    ).toMatchSnapshot()
  })

  it('calls onClickOutside when we click on the wrapper area', () => {
    const modal = mount(createComponentWithProps(Modal, props))
    const wrapper = modal.find('.modal-wrapper')
    wrapper.simulate('click')
    expect(props.onClickOutside).toHaveBeenCalledTimes(1)
  })

  it('does not call onClickOutside when we click on the content area', () => {
    const modal = mount(createComponentWithProps(Modal, props))
    const content = modal.find('.modal-content')
    content.simulate('click')
    expect(props.onClickOutside).toHaveBeenCalledTimes(0)
  })
})
