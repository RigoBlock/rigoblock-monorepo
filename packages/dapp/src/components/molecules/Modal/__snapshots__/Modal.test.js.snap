// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders correctly 1`] = `
<div
  className="modal-wrapper"
  onClick={[Function]}
>
  <div
    className="modal-content"
  >
    <div>
       hello
    </div>
  </div>
</div>
`;
